video_xblock/tests/test_backends.py

```diff
@@ -0,0 +1,253 @@
+        'https://studio.brightcove.com/products/videocloud/media/videos/45263567468485',
+        'https://wi.st/medias/HRrr784kH8932Z',
+        'https://vimeo.com/202889234'
+    ]
+
+    auth_mocks = [
+        YoutubeAuthMock,
+        BrightcoveAuthMock,
+        WistiaAuthMock,
+        VimeoAuthMock,
+    ]
+
+    default_trans_mocks = [
+        YoutubeDefaultTranscriptsMock,
+        BrightcoveDefaultTranscriptsMock,
+        WistiaDefaultTranscriptsMock,
+        VimeoDefaultTranscriptsMock,
+    ]
+
+    download_transcript_mocks = [
+        YoutubeDownloadTranscriptMock,
+        BrightcoveDownloadTranscriptMock,
+        WistiaDownloadTranscriptMock,
+        VimeoDownloadTranscriptMock,
+    ]
+
+    @XBlock.register_temp_plugin(brightcove.BrightcovePlayer, 'brightcove')
+    @XBlock.register_temp_plugin(wistia.WistiaPlayer, 'wistia')
+    @XBlock.register_temp_plugin(youtube.YoutubePlayer, 'youtube')
+    @XBlock.register_temp_plugin(vimeo.VimeoPlayer, 'vimeo')
+    def setUp(self):
+        super(TestCustomBackends, self).setUp()
+        self.player = {}
+        for backend in self.backends:
+            player_class = XBlock.load_class(backend)
+            self.player[backend] = player_class
+
+    def test_get_player_html(self):
+        """
+        Check that player files content is returned in Response body.
+        """
+        context = {
+            'player_state': {
+                'transcripts': [{
+                    'lang': 'en',
+                    'label': 'English',
+                    'url': 'http://test.url'
+                }],
+                'current_time': ''
+            },
+            'url': '',
+            'start_time': '',
+            'end_time': ''
+        }
+        for backend in self.backends:
+            player = self.player[backend]
+            res = player(self.xblock).get_player_html(**context)
+            self.assertIn('window.videojs', res.body)
+
+    @data(*zip(backends, media_urls))
+    @unpack
+    def test_match(self, backend, url):
+        """
+        Check if provided video `href` validates in right way.
+        """
+        player = self.player[backend]
+        res = player.match(url)
+        self.assertTrue(bool(res))
+
+        # test wrong data
+        res = player.match('http://wrong.url')
+        self.assertFalse(bool(res))
+
+    @data(
+        ([{'lang': 'ru'}], [{'lang': 'en'}, {'lang': 'uk'}]),
+        ([{'lang': 'en'}, {'lang': 'uk'}], [{'lang': 'ru'}]),
+        ([{'lang': 'some_other_lng'}], [{'lang': 'en'}, {'lang': 'ru'}, {'lang': 'uk'}])
+    )
+    @unpack
+    def test_filter_default_transcripts(self, transcripts, default):
+        """
+        Check transcripts are excluded from the list of available ones in video xblock.
+        """
+        for backend in self.backends:
+            player = self.player[backend](self.xblock)
+            default_transcripts = [{'lang': 'en'}, {'lang': 'ru'}, {'lang': 'uk'}]
```

Comment:
Move this line out of loop, so it's more obvious `default_transcripts` is the same for all cases.